File: backend/src/main/java/net/zorphy/backend/main/game/dto/GameFilters.java
package net.zorphy.backend.main.game.dto;


import java.time.Duration;
import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record GameFilters(
        String text,
        Instant dateFrom,
        Instant dateTo,
        Integer minPlayers,
        Integer maxPlayers,
        Duration minDuration,
        Duration maxDuration,
        List<GameType> gameTypes,
        List<UUID> players
) {
}

File: frontend/src/app/main/games/dto/GameFilters.ts
import {GameType} from "./GameType";

export interface GameFilters {
    text: string | null;
    dateFrom: string | null,
    dateTo: string | null,
    minPlayers: number | null,
    maxPlayers: number | null,
    minDuration: string | null,
    maxDuration: string | null,
    gameTypes: GameType[] | null,
    players: string[] | null
}